### PR TITLE
Improve mapping of MDBX error codes to human-readable errors

### LINKED ISSUE #3687: Improve mapping of MDBX error codes to human-readable errors
### Describe the feature

Opening the MDBX database file throws file system error codes that are not mapped, so they end up propagated as the error code. So far I've came across error 37 (under WSL1 only) and 13 (permission denied, solved with a change of user) which where very hard to figure out as they are not in the libmdbx code and I'm not a C developer so I'm not familiar with the codes. Digging a bit online I found [this list](https://www.thegeekstuff.com/2010/10/linux-error-codes/) which could be the right list of errors to include, though someone else more familiar with MDBX would know better than me. If the list is correct, I could take the task of mapping such errors and create a PR.

A note on error 37: it's not evident that this is thrown because the OS is WSL1, so it might be good to include a note when thrown noting that mdbx doesn't work on WSL1 but it does on WSL2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #3687.

### CODE REVIEW

File: crates/storage/libmdbx-rs/src/error.rs [hunk 1]
```diff
@@ -1,70 +1,116 @@
...
 /// An MDBX error kind.
 #[derive(Debug, thiserror::Error, Clone, PartialEq, Eq)]
 pub enum Error {
-    /// The key/value pair already exists.
+    /// Key/data pair already exists.
+    #[error("key/data pair already exists")]
     KeyExist,
-    /// The requested key/value pair was not found.
+    /// No matching key/data pair found.
+    #[error("ko matching key/data pair found")]
```
Comment (rkrasiuk) [inline crates/storage/libmdbx-rs/src/error.rs:14]:
```suggestion
    #[error("no matching key/data pair found")]
```

**Review by rkrasiuk — APPROVED** [review]

File: crates/storage/libmdbx-rs/src/error.rs [hunk 2]
```diff
@@ -1,70 +1,116 @@
...
+    #[error("page has no more space")]
     PageFull,
-    /// The database engine was unable to extend mapping, e.g. the address space is unavailable or
-    /// busy.
-    ///
-    /// This can mean:
-    /// - The database size was extended by other processes beyond the environment map size, and
-    ///   the engine was unable to extend the mapping while starting a read transaction. The
-    ///   environment should be re-opened to continue.
-    /// - The engine was unable to extend the mapping during a write transaction or an explicit
```
Comment (Rjected) [inline crates/storage/libmdbx-rs/src/error.rs:43]: can we keep this comment for this variant?

**Review by Rjected — COMMENTED** [review]
just one note about comments, otherwise this looks good to me

File: crates/storage/libmdbx-rs/src/error.rs (hunk 2 above)
Comment (tcoratger) [reply crates/storage/libmdbx-rs/src/error.rs]: Sure I've just added it back

**Review by Rjected — APPROVED** [review]
looks good to me, thanks!

**Pull request merged**